src/io/keyboard.js

```diff
@@ -16,87 +39,111 @@ class Keyboard {
     }
 
     /**
-     * Convert a Scratch key name to a DOM keyCode.
-     * @param {Any} keyName Scratch key argument.
-     * @return {number} Key code corresponding to a DOM event.
-     * @private
+     * Convert from a keyboard event key name to a Scratch key name.
+     * @param  {string} keyString the input key string.
+     * @return {string} the corresponding Scratch key, or an empty string.
      */
-    _scratchKeyToKeyCode (keyName) {
-        if (typeof keyName === 'number') {
-            // Key codes placed in with number blocks.
-            return keyName;
-        }
-        const keyString = Cast.toString(keyName);
+    _keyStringToScratchKey (keyString) {
+        keyString = Cast.toString(keyString);
+        // Convert space and arrow keys to their Scratch key names.
         switch (keyString) {
-        case 'space': return 32;
-        case 'left arrow': return 37;
-        case 'up arrow': return 38;
-        case 'right arrow': return 39;
-        case 'down arrow': return 40;
-        // @todo: Consider adding other special keys here.
+        case ' ': return KEY_NAME.SPACE;
+        case 'ArrowLeft':
+        case 'Left': return KEY_NAME.LEFT;
+        case 'ArrowUp':
+        case 'Up': return KEY_NAME.UP;
+        case 'Right':
+        case 'ArrowRight': return KEY_NAME.RIGHT;
+        case 'Down':
+        case 'ArrowDown': return KEY_NAME.DOWN;
+        }
+        // Ignore modifier keys
```

Comment:
Are modifier keys always > length 1 and all things > length 1 are necessarily modifier keys?

Reply:
Those both seem to be generally true (except for the special cases handled just before that line). 

MDN's list of modifier keys is here: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values

There are things > 1 in other languages that "look like" single characters- the question is whether they can ever be passed as single key press events (not sure). For now we are not supporting these, but something to look out for in the future.

Reply:
This is awesome! Thanks for the link =]